_templates/component/new/index.ejs.t

```diff
@@ -5,14 +5,22 @@ to: <%= app_name %>/<%= path %>/<%= name %>/index.tsx
 
 import React from 'react';
 import styled from 'styled-components';
+<%= isPayloadBlock && `import { Page } from '@web/payload/payload-types';` %>
 
 const Container = styled.div``;
 
-export type <%= name %>Type = {
+<%= isPayloadBlock ? `export type ExtractBlockType<T, BlockType> = T extends { blockType: BlockType }
+  ? T
+  : never;
+export type <%= name %>Type = ExtractBlockType<
+  NonNullable<Page['layout']>[number],
+  'ImageWithContent'
```

Comment:
Not sure why this is hardcoded or what it's doing. 

Not sure why this is importing `Page` -- shouldn't it import the block type itself?

If either of the above are too much, just skip the plop all together